File: replace-colors.py
import re
import sys

file_path = r'src\pages\Scenarios.jsx'

try:
    with open(file_path, 'r', encoding='utf-8') as f:
        content = f.read()
except Exception as e:
    print(f'Error reading file: {e}')
    sys.exit(1)

# Order matters: do hover: and shadow: first to avoid double replacements
replacements = [
    ('hover:bg-[#ffffff]/10', 'hover:bg-muted'),
    ('hover:shadow-[#ffffff]/10', 'hover:shadow-foreground/10'),
    ('shadow-[#ffffff]/20', 'shadow-foreground/20'),
    ('bg-[#0E0E0E]', 'bg-card'),
    ('border-[#ffffff]/50', 'border-foreground/50'),
    ('border-[#ffffff]/40', 'border-foreground/40'),
    ('border-[#ffffff]/20', 'border-foreground/20'),
    ('border-white/10', 'border-border'),
    ('bg-[#ffffff]/10', 'bg-muted'),
    ('ring-white/10', 'ring-border')
]

print('\nReplacement Report:')
print('===================')

total = 0
for old, new in replacements:
    escaped = re.escape(old)
    count = len(re.findall(escaped, content))
    content = re.sub(escaped, new, content)
    print(f'{count} × {old} → {new}')
    total += count

try:
    with open(file_path, 'w', encoding='utf-8') as f:
        f.write(content)
    print(f'\nTotal replacements: {total}')
    print('\nFile updated successfully!')
except Exception as e:
    print(f'\nError writing file: {e}')
    sys.exit(1)

File: replace-colors.js
const fs = require('fs');
const path = require('path');

const filePath = path.join(__dirname, 'src', 'pages', 'Scenarios.jsx');
let content = fs.readFileSync(filePath, 'utf8');

const replacements = [
  { from: 'bg-[#0E0E0E]', to: 'bg-card', count: 0 },
  { from: 'border-white/10', to: 'border-border', count: 0 },
  { from: 'border-[#ffffff]/20', to: 'border-foreground/20', count: 0 },
  { from: 'border-[#ffffff]/40', to: 'border-foreground/40', count: 0 },
  { from: 'border-[#ffffff]/50', to: 'border-foreground/50', count: 0 },
  { from: 'bg-[#ffffff]/10', to: 'bg-muted', count: 0 },
  { from: 'hover:bg-[#ffffff]/10', to: 'hover:bg-muted', count: 0 },
  { from: 'hover:shadow-[#ffffff]/10', to: 'hover:shadow-foreground/10', count: 0 },
  { from: 'shadow-[#ffffff]/20', to: 'shadow-foreground/20', count: 0 },
  { from: 'ring-white/10', to: 'ring-border', count: 0 }
];

for (let replacement of replacements) {
  const regex = new RegExp(replacement.from.replace(/[.*+?^${}()|[\]\\]/g, '\\$&'), 'g');
  const matches = content.match(regex);
  replacement.count = matches ? matches.length : 0;
  content = content.replace(regex, replacement.to);
}

fs.writeFileSync(filePath, content, 'utf8');

console.log('\nReplacement Report:');
console.log('===================');
for (let replacement of replacements) {
  console.log(`${replacement.count} × ${replacement.from} → ${replacement.to}`);
}
console.log('\nTotal replacements:', replacements.reduce((sum, r) => sum + r.count, 0));
